### DISCUSSION TITLE
@babel/parser: when I use parseExpression to parse code, I get some errors.

### DISCUSSION THREAD

**Discussion by Losstie** [babel/babel#15045] [Q&A] [answered]
I want to use parseExpression of @babel/parser to parse some Expression, but I got some SYNTAX_ERROR! 

The code looks like this：
`
const parser = require("@babel/parser");
const code = "const a =2";
 parser.parseExpression(code, {
      sourceType: "module",
        errorRecovery: true,
        plugins: ["typescript",  "decorators", "jsx"],
        });
`
and, I got this error: SYNTAX_ERROR:UnexpectedToken!

Am I using it wrong or is this function just not able to parse this kind of statement?

**Comment by liuxingbaoyu** [comment]
`const` is not an `Expression` but a `Statement`, please use `parse()`.

**Comment by nicolo-ribaudo** [comment, marked as answer]
An expression is something that would be valid when wrapped in `(`...`)`. For example, `1 + 2` is an expression because `(1 + 2)` is valid code. `const a = 1` isn't an expression, because `(const a = 1)` is not valid JavaScript.

**Reply by Losstie** [reply]
thanks～
